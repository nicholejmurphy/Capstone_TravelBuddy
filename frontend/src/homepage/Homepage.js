import React, { useContext } from "react";
// import "./Homepage.css";
import {
  Button,
  CardBody,
  // CardSubtitle,
  CardText,
  // CardTitle,
} from "reactstrap";
import UserContext from "../auth/UserContext";

function Homepage() {
  const { currUser } = useContext(UserContext);

  function isLoggedIn() {
    return (
      <CardBody className="Homepage bg-light shadow rounded">
        <CardText className="Homepage-welcome display-4">
          Welcome back, {currUser.firstName}!
        </CardText>
      </CardBody>
    );
  }

  function notLoggedIn() {
    return (
      <CardBody className="Homepage bg-light">
        <CardText className="Homepage-btns text-center">
          <Button className="bg-primary" tag="a" href="/login">
            Login
          </Button>
          <Button className="bg-primary" tag="a" href="/signup">
            SignUp
          </Button>
        </CardText>
      </CardBody>
    );
  }

  return currUser ? isLoggedIn() : notLoggedIn();
}

export default Homepage;
