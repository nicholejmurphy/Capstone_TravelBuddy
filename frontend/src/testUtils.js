import React from "react";
import UserContext from "./auth/UserContext";

const testUser = {
  id: 234,
  username: "testuser",
  first_name: "testuser",
  last_name: "testuser",
  location_id: "60489",
  location_name: "Asheville, NC",
};

const UserProvider = (currUser = testUser, { children }) => (
  <UserContext.Provider value={{ currUser }}>{children}</UserContext.Provider>
);

export { UserProvider };
