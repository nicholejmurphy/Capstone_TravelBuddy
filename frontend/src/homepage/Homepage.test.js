import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import Homepage from "./Homepage";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot w/ no user logged in", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
