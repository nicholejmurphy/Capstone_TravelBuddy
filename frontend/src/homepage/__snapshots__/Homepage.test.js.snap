// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<DocumentFragment />`;

exports[`matches snapshot w/ no user logged in 1`] = `<DocumentFragment />`;
